pyrit/prompt_target/crescendo_target.py

```diff
@@ -0,0 +1,214 @@
+                    {
+                        "category": "HARM_CATEGORY_DANGEROUS",
+                        "threshold": "BLOCK_NONE",
+                    }
+                ]
+            else:
+                 self._GeminiSafetySettings = GeminiSafetySettings
+            self._llm = getGeminiPromptInstance(api_key=api_key,api_deployment_name=api_deployment_name)
+        elif llmName == "Claude":
+            self._llm = getClaudeInstance(api_key=api_key,api_deployment_name=api_deployment_name)
+        else:
+            raise ValueError("Invalid LLM name")
+
+        # Prompt templates
+
+
+    # Adapting the chat history to the Llama-2 format
+    def langhchain_to_llama2(inputObject,Llama2Tokenizer):
+        history = Llama2Tokenizer.eos_token
+        for item in inputObject:
+            if item.type == 'human':
+                history += "[INST]" + item.content + "[/INST]"
+            else:
+                history += item.content + Llama2Tokenizer.eos_token
+        tokens = Llama2Tokenizer(history, return_tensors="pt", add_special_tokens=False)
+        return history, tokens['input_ids']
+
+    # Parsing the output of the LLM
+    def getParsedContent(self, template, elementNames): 
+        llm = self._llm
+        counter = 0
+        # Trying to parse the output 5 times else returning an error
+        while counter < 5:  
+            counter = counter+ 1
+            try:  
+                
+                flag,output = self.callLLM([HumanMessage(content=template)])
+                if not flag:
+                    print("Error: %s"%output)
+                    return flag,output 
+                
+                parsed_output = json.loads(output.content)  
+                parsed_elements = {}  
+                
+                for elementName in elementNames:  
+                    parsedElement = parsed_output.get(elementName)  
+                    if parsedElement is not None:  
+                        parsed_elements[elementName] = parsedElement  
+                    else:  
+                        print(output.content)  
+                        print(f"Error: '{elementName}' key not found in the output")  
+                        break  
+                else:  
+                    return True,parsed_elements  
+    
+            except json.JSONDecodeError as e:  
+                print(output.content)  
+                print("Error: Incorrect JSON format", str(e))  
+                continue  
+
+        return False,'Error: Incorrect JSON format'
+
+    # Adapting the chat history to the Gimini format
+    def langchain_to_gemini(inputObject):
+        history = [
+            {
+                "role": "user" if item.type == 'human' else "model",
+                "parts": [{"text": item.content}],
+            } 
+            for item in inputObject
+        ]
+        return history
+
+    # Calling the LLM. Adapting the input format depending on the LLM
+    def callLLM(self, inputObject):
```

Comment:
```suggestion
    def call_llm(self, inputObject):
```
nit: method names should be snake case